### PR TITLE
 Added userMeta method.

### PR DESCRIPTION
**Pull request by sohda**
## Changes
- Added MediaStorage#userMeta.
- Added MediaStorageRequest#put.
- Modified MediaStorageRequest#generateRequest.
## Tests
- Show here: [Test Result After 2nd Review](https://pflabo.atlassian.net/browse/NECO-164#After2ndReview)

### CODE REVIEW

File: Source/MediaStorage.swift [hunk 1]
```diff
@@ -236,7 +239,52 @@ public class MediaStorage {
...
+    
+    public func addMeta(mediaId mediaId: String, userMeta: Dictionary<String, String>, completionHandler: MediaStorageError -> Void) {
+        if accessToken == nil {
+            completionHandler(
+                MediaStorageError(statusCode: nil, message: "wrong usage: use the connect method to get an access token.")
+            )
+            return
+        }
+        
+        for (key, value) in userMeta {
```
Comment (YukiThornton) [inline Source/MediaStorage.swift:257]: I think this method should return a `completionHandler` with or without some error message if `userMeta` is `nil` or an empty `Dictionary`. When `userMeta` is `nil` or empty, the process does not go through this loop.

File: Source/MediaStorage.swift [hunk 2]
```diff
@@ -236,7 +239,52 @@ public class MediaStorage {
...
+            let userMetaKey = replaceUserMeta(key)
+            if userMetaKey.isEmpty || !isValidValue(value) {
+                completionHandler(
+                    MediaStorageError(statusCode: nil, message: "invalid parameter: \(userMeta)")
```
Comment (YukiThornton) [inline Source/MediaStorage.swift]: I think it'd be easier to read the error message if it has the single key&value set which is invalid, not the whole sets in `userMeta`.

File: Source/MediaStorage.swift (hunk 1 above)
Comment (sohda) [inline Source/MediaStorage.swift:257]: Thank you for notice it.
I'm going to add following code at the above the `for (...` sentence.

``` swift
        if userMeta.isEmpty {
            completionHandler(
                MediaStorageError(statusCode: nil, message: "empty userMeta was taken: nothing to request.")
            )
        }
```

File: Source/MediaStorage.swift (hunk 2 above)
Comment (sohda) [inline Source/MediaStorage.swift]: Okay, I'm going to fix it 👍

File: Source/MediaStorage.swift [hunk 3]
```diff
@@ -236,7 +239,58 @@ public class MediaStorage {
...
+        if accessToken == nil {
+            completionHandler(
+                MediaStorageError(statusCode: nil, message: "wrong usage: use the connect method to get an access token.")
+            )
+            return
+        }
+        
+        if userMeta.isEmpty {
+            completionHandler(
+                MediaStorageError(statusCode: nil, message: "empty userMeta was taken: nothing to request.")
```
Comment (YukiThornton) [inline Source/MediaStorage.swift]: To simplify the message, how about this one; "userMeta is empty: nothing to request."

**Comment by YukiThornton** [comment]
I reviewed all the changes and they seem great! So I'm gonna merge this pull request.

**Pull request merged**